samples/GStreamerWPF/GStreamerD3D11/GStreamerCanvas.xaml.cs

```
@@ -62,25 +73,38 @@ private void Control_Loaded(object sender, RoutedEventArgs e)
 
         private void Reload()
         {
-            if (!Enabled)
+            lock (_lockObject)
             {
-                return;
+                if (!Enabled)
+                {
+                    return;
+                }
+
+                try
+                {
+                    _D3DImageEx = new D3DImageEx();
+                    _D3D11Scene = new D3D11TestScene(1920, 1080);
+
+                    d3dScene.Source = _D3DImageEx;
+
+                    /* Set the backbuffer, which is a ID3D11Texture2D pointer */
+                    var renderTarget = _D3D11Scene.GetRenderTarget();
+                    var backBuffer = _D3DImageEx.CreateBackBuffer(D3DResourceTypeEx.ID3D11Texture2D, renderTarget);
+                    var enableSoftwareFallback = false;
+                    _D3DImageEx.Lock();
+                    _D3DImageEx.SetBackBuffer(D3DResourceType.IDirect3DSurface9, backBuffer, enableSoftwareFallback);
+                    _D3DImageEx.Unlock();
+
+                    CompositionTarget.Rendering += CompositionTarget_Rendering;
+                }
+                catch (Exception ex)
+                {
+                    DisposeScene();
+                    Enabled = false;
+                    txtError.Visibility = Visibility.Visible;
```

Does this ever have to be hidden again?

---

no if we get this error this component will stay in error so no need to hide error message again.